### ISSUE TITLE
[Entity Masker] Massive performance cost

### ISSUE THREAD

**Issue by fuzxi** [Droei/Pet-The-Capybara#1]
With the plugin enabled, all settings unchecked, and the NPC list empty, my frame rate at the GE is hovering between 20-40 FPS. Disabling the plugin, my FPS immediately jumps up to 144 and locks there. Why is there such a huge performance hit when it's not even masking any NPCs?

**Comment by Droei** [comment]
That's weird, I myself haven't noticed that yet.
I'll look into it thanks for letting me know!

**Comment by IllumiC** [comment]
Recreated with this .properties file (+default entity masker without jad images)
[Liam.txt](https://github.com/user-attachments/files/16320008/Liam.txt)
10fps north lumbridge chickens, 20fps lumbridge castle.

Edit:
But I must say that I love the plugin

**Comment by peakjcw** [comment]
I'm confirming the same issue. Even with nothing to mask, enabling the plugin drops my FPS from 60+ to 10-20. My FPS sticks at 60 everywhere from Fossil Island to the GE on W302 unless Entity Masker is on.
The issue gradually gets worse over the course of 30 minutes or so from starting up RuneLite.

It's a shame too, because seeing bugs even in OSRS makes me very ill and this plugin was a lifesaver that I very much appreciate. At least I can toggle it on when really needed.

**Comment by Droei** [comment]
Hi there! Thanks for confirming, I also got it confirmed by others in private too and I could recreate the problem on my device!
I've been very busy lately so I couldn't get to it yet but I am planning to update all my plugins soon.

Also on a side @peakjcw if you want me to I can take your list of bugs and put it into the plugin as a default option together with the other phobias!

I will get back to this problem soon and keep y'all updated! 
Thanks for using my plugin!

**Comment by Droei** [comment]
Hi everyone!

Sorry that I've been this late with this, life has been busy and I procrastonated a little.
I have found the issue, the performance drop comes from the jad images, disabling Show Images should resolve the issue and I will push out a new update soon to temporary disable the option by default.
Once things settle down a little on my side I hope to redo that code to be more performance friendy and customisable!

Thanks for your patience and for using my plugin!

**Comment by IllumiC** [comment]
Hey, I've been experiencing performance drops with the Jad images disabled.

**Comment by peakjcw** [comment]
Yeah I should have clarified before but I was always having issues with the images off.

**Comment by Droei** [comment]
Damn alright I also read that indeed everything was off.
Sorry about that, I could only recreate the issue when I was messing with the images.

I'd love to hear what people experiencing these problems use in their desktop: CPU, GPU & Ram.
Asking this to check if its anything hardware related

**Comment by IllumiC** [comment]
I'm on a 11700k, 3060 12gb, 16gb ram.
My girlfriend experiences the issue in her similar amd system. Haven't been
on osrs much but next time I'm on I'll have a go seeing if it's any other
plugin causing it.

**Comment by Droei** [comment]
That is better than my current setup, I used to have a high end setup but now its 6y old and I'm having a lot of problems with it lately.
But for some reason it just doesn't lagg with entity masker (unless images are on) so maybe it is indeed a plugin thing.

Still definitly something that needs to be resolved so I'll take any info I can get!

**Comment by uqx7** [comment]
Definitely can still recreate this one. was having performance drops after 1hr~ gameplay last few days. low 10~ fps in lumbridge, 30fps in farming guild. when turning off the plugin FPS shoots back up to 50 (uncapped it goes to 200+)
My partner can replicate the issue too, performance isn't suffering until after 1hr or more. 
![entitymaskersettings](https://github.com/user-attachments/assets/ffd18470-a226-4521-bff2-c95df864cbb4)
